src/veh_interact.cpp

```diff
@@ -2297,8 +2255,8 @@ void complete_vehicle ()
         if (!broken) {
             g->m.add_item_or_charges( g->u.pos(), veh->parts[vehicle_part].properties_to_item() );
             // simple tasks won't train mechanics
-            if(type != SEL_JACK && !is_hand_remove) {
-                g->u.practice( skill_mechanics, (is_wood || is_wrenchable || is_screwable) ? 15 : 30);
+            if( reqs.get_tools().empty() ) {
+                g->u.practice( skill_mechanics, vpinfo.difficulty * 50 );
```

Comment:
Looks like inverted logic here:
Before, it would train mechanics when it consumed tools.
Now it trains it when it doesn't require tools.

Also a moderate balance change.
Should probably be tied to install/remove time rather than difficulty. There is no risk of messing up, so installing and removing a complex part over and over shouldn't train the skill more than the same grind with a simple part.